src/Utils.php

```diff
@@ -0,0 +1,840 @@
+        print_r($var);
+        echo "</pre>";
+    }
+
+    public static function tlms_pre($var): void
+    {
+        echo "<pre>";
+        print_r($var);
+        echo "</pre>";
+        exit;
+    }
+
+    public static function tlms_vd($var): void
+    {
+        echo "<pre>";
+        var_dump($var);
+        echo "</pre>";
+    }
+
+    public static function tlms_limitWords(string $string, int $limit): string
+    {
+        if ($limit) {
+            $words = explode(" ", $string);
+
+            return implode(
+                " ",
+                array_splice($words, 0, $limit)
+            );
+        } else {
+            return $string;
+        }
+    }
+
+    public static function tlms_limitSentence(string $string, int $limit): string
+    {
+        $sentences = explode(".", $string);
+
+        return implode(".", array_splice($sentences, 0, $limit));
+    }
+
+    public static function tlms_isValidDomain(string $domain): bool
+    {
+        return preg_match("/^[a-z0-9-\.]{1,100}\w+$/", $domain)
+            and (strpos($domain, 'talentlms.com') !== false);
+    }
+
+    public static function tlms_isApiKey(string $apiKey): bool
+    {
+        if (strlen($apiKey) === 30) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public static function tlms_parseDate(string $format, string $date): DateTime
+    {
+        $isPM = (stripos($date, 'PM') !== false);
+        $parsedDate = str_replace(array('AM', 'PM'), '', $date);
+        $is12hourFormat = ($parsedDate !== $date);
+        $parsedDate = DateTime::createFromFormat(trim($format), trim($parsedDate));
+
+        if ($is12hourFormat) {
+            if ($isPM && $parsedDate->format('H') !== '12') {
+                $parsedDate->modify('+12 hours');
+            } elseif (!$isPM && $parsedDate->format('H') === '12') {
+                $parsedDate->modify('-12 hours');
+            }
+        }
+
+        return $parsedDate;
+    }
+
+    public static function tlms_getDateFormat(bool $no_sec = false): string
+    {
+        try {
+            $site_info = self::tlms_getTalentLMSSiteInfo();
+            $date_format = $site_info['date_format'];
+        } catch (Exception $exception) {
+            $date_format = '';
+        }
+
+        switch ($date_format) {
+            case 'DDMMYYYY':
+                if ($no_sec) {
+                    $format = 'd/m/Y';
+                } else {
+                    $format = 'd/m/Y, H:i:s';
+                }
+                break;
+            case 'MMDDYYYY':
+                if ($no_sec) {
+                    $format = 'm/d/Y';
+                } else {
+                    $format = 'm/d/Y, H:i:s';
+                }
+                break;
+            case 'YYYYMMDD':
+            default:
+                if ($no_sec) {
+                    $format = 'Y/m/d';
+                } else {
+                    $format = 'Y/m/d, H:i:s';
+                }
+                break;
+        }
```

Comment:
We also have a 12-hour format in the domain settings. My guess is that we want this together with our date formatting.

Reply:
@ypanagopoulos Is this proposal in the scope of the refactor task? Should we create a new task to add this?

Reply:
Yes lets do this on a different task. @DimChristodoulou can you create it?